### PR TITLE
HPCC-16901 ECLCC compile/link error if folding overflows

### PR DESCRIPTION
**Pull request by richardkchapman**
Also first stab at rationalizing the divideByZero flag to apply to other
functions such as SQRT(-1).

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Not ready for pulling, but I would like your comments on the direction before I push it further

File: common/deftype/defvalue.cpp [hunk 1]
```diff
@@ -2022,12 +2022,18 @@ unsigned RealValue::getHash(unsigned initval)
 
 const char *RealValue::generateECL(StringBuffer &s)
 {
-    return getStringValue(s);
+    if (isinf(val))
+        return s.append("INFINITY");
```
Comment (richardkchapman) [inline common/deftype/defvalue.cpp]: Not sure what generateECL is used for - I was thinking we might introduce a new keyword INFINITY (and maybe one for NAN too)...

**Comment by richardkchapman** [comment]
@johnholt FYI
@ghalliday I think this is ready for review.

If accepted needs a Jira opened to document the new Std.Math module and update the docs on the functions that are now affected by the DivideByZero flag

**Comment by richardkchapman** [comment]
@AttilaVamos The smoketest failure here looks like a false positive to me - not sure what is going on but maybe picking up the std library from the wrong place ?

**Comment by AttilaVamos** [comment]
@richardkchapman I'm checking it

**Comment by richardkchapman** [comment]
@ghalliday Please review (smoketest issues have been resolved)

**Pull request merged**